cluster/src/main/java/org/apache/iotdb/cluster/utils/ClusterUtils.java

```diff
@@ -368,6 +376,90 @@ public static PartitionGroup partitionByPathTimeWithSync(
     return partitionGroup;
   }
 
+  public static void setVersionForSpecialPlan(PhysicalPlan plan, long currLogIndex)
+      throws MetadataException {
+    long majorVersion = 0;
+    long minorVersion = 0;
+    if (plan.getOperatorType() == null) {
+      logger.warn("the plan={} have no operator, ignore it", plan);
+      return;
+    }
+    switch (plan.getOperatorType()) {
+      case SET_STORAGE_GROUP:
+        majorVersion = currLogIndex;
+        minorVersion = 0;
+        plan.setMajorVersion(majorVersion);
+        plan.setMinorVersion(minorVersion);
+        break;
+      case DELETE_STORAGE_GROUP:
+        List<PartialPath> deletePathList = new ArrayList<>();
+        for (PartialPath storageGroupPath : plan.getPaths()) {
+          List<PartialPath> allRelatedStorageGroupPath =
+              IoTDB.metaManager.getStorageGroupPaths(storageGroupPath);
+          if (allRelatedStorageGroupPath.isEmpty()) {
+            throw new PathNotExistException(storageGroupPath.getFullPath(), true);
+          }
+          for (PartialPath path : allRelatedStorageGroupPath) {
+            StorageGroupMNode storageGroupMNode = IoTDB.metaManager.getStorageGroupNodeByPath(path);
+            majorVersion = storageGroupMNode.getMajorVersion();
+            minorVersion = storageGroupMNode.getMinorVersion();
+            path.setMajorVersion(majorVersion);
+            path.setMinorVersion(minorVersion);
+            deletePathList.add(path);
+          }
+        }
+        // replace the to be deleted paths
+        ((DeleteStorageGroupPlan) plan).setPaths(deletePathList);
+        break;
+      case CREATE_TIMESERIES:
+        PartialPath partialPath = ((CreateTimeSeriesPlan) plan).getPath();
+        StorageGroupMNode storageGroupMNode =
+            IoTDB.metaManager.getStorageGroupNodeByPath(partialPath);
+        majorVersion = storageGroupMNode.getMajorVersion();
+        minorVersion = currLogIndex;
+        ((CreateTimeSeriesPlan) plan).getPath().setMajorVersion(majorVersion);
+        ((CreateTimeSeriesPlan) plan).getPath().setMinorVersion(minorVersion);
+        break;
+      case ALTER_TIMESERIES:
```

Comment:
So, why do not check version when we alter timeseries?

Reply:
Thanks for your carefully review. I have add it.